### ISSUE TITLE
Failure of autodetecting mobile devices.

### ISSUE THREAD

**Issue by yasuhirokimura** [RainLoop/rainloop-webmail#1507]
Server OS: FreeBSD 11.1-RELEASE amd64
PHP: 7.1 (Installed using FreeBSD ports)
RainLoop: v 1.11.2 Community Edition (Installed using FreeBSD ports)

RainLoop autodetects mobile devices and works with mobile version if detected.
But there are some cases autodetection does not work correctly.
For example, I have following 3 mobile devices.

- Google Nexus 7 (2013) (Android 6.0.1)
- Google Nexus 6p (Android 8.0.0)
- ASUS ZenPad 3 (Android 7.0)

Latest versions of Chrome and Firefox are installed to all of them. And with 2*3=6 combinations of devices and browsers RainLoop works as followings.

Device and Browser  | Which version RainLoop works with
-----------------------|--------------------------------------
Nexus 7 + Chrome   | Desktop version
Nexus 7 + Firefox     | Desktop version
Nexus 6p + Chrome | Mobile version
Nexus 6p + Firefox   | Mobile version
ZenPad 3 + Chrome | Mobile version
ZenPad 3 + Firefox   | Desktop version

While I hope RainLoop detects all cases correctly, It may be necessary to always provide menu items to switch between mobile and desktop version as a workaround of this issue.

**Comment by tomasz-c** [comment]
The MobileDetect module is not up to date, despite the information in CHANGELOG:
- Update all third party libraries

The MobileDetect version in Rainloop is 2.8.22 out of 24 Apr 2016, and the last one is 2.8.26 out of 23 days ago.
Please upgrade, this improves the detection of multiple devices.

@yasuhirokimura:
Please check the latest version of MobileDetect if devices are recognized correctly:
http://demo.mobiledetect.net/

**Comment by yasuhirokimura** [comment]
I checked above 6 combinations with http://demo.mobiledetect.net/ and got following results.

| Device and Browser | This is a ... |
------------------------|------------|
| Nexus 7 + Chrome | tablet |
| Nexus 7 + Firefox | tablet |
| Nexus 6p + Chrome | phone |
| Nexus 6p + Firefox | phone |
| ZenPad 3 + Chrome | phone |
| ZenPad 3 + Firefox | tablet |

Does this mean RainLoop works with desktop version if device is detected as 'tablet'?

**Comment by RainLoop** [comment]
> Does this mean RainLoop works with desktop version if device is detected as 'tablet'?

Yep. [Service.php#L160](https://github.com/RainLoop/rainloop-webmail/blob/master/rainloop/v/0.0.0/app/libraries/RainLoop/Service.php#L160)

**Comment by yasuhirokimura** [comment]
For me 7 or 8 inch displays are not wide enough to use RainLoop with desktop version. So would you please consider one of followings?

1. Change behavior when device is detected as tablet.
2. Make it possible to customize which version is used for tablet in admin panel.
3. Provide menu item to switch from desktop to mobile.

(In preferable order for me)
